bitcorenode/index.js

```diff
@@ -0,0 +1,202 @@
+'use strict';
+
+var util = require('util');
+var fs = require('fs');
+var io = require('socket.io');
+var https = require('https');
+var http = require('http');
+var async = require('async');
+var path = require('path');
+var bitcore = require('bitcore');
+var Networks = bitcore.Networks;
+var Locker = require('locker-server');
+var BlockchainMonitor = require('../lib/blockchainmonitor');
+var EmailService = require('../lib/emailservice');
+var ExpressApp = require('../lib/expressapp');
+var WsApp = require('../lib/wsapp');
+var child_process = require('child_process');
+var spawn = child_process.spawn;
+var EventEmitter = require('events').EventEmitter;
+var baseConfig = require('../config');
+
+/**
+ * A Bitcore Node Service module
+ * @param {Object} options
+ * @param {Node} options.node - A reference to the Bitcore Node instance
+-* @param {Boolean} options.https - Enable https for this module, defaults to node settings.
+ * @param {Number} options.bwsPort - Port for Bitcore Wallet Service API
+ * @param {Number} options.messageBrokerPort - Port for BWS message broker
+ * @param {Number} options.lockerPort - Port for BWS locker port
+ */
+var Service = function(options) {
+  EventEmitter.call(this);
+
+  this.node = options.node;
+  this.https = options.https || this.node.https;
+  this.httpsOptions = options.httpsOptions || this.node.httpsOptions;
+  this.bwsPort = options.bwsPort || Service.BWS_PORT;
+  this.messageBrokerPort = options.messageBrokerPort || Service.MESSAGE_BROKER_PORT;
+  this.lockerPort = options.lockerPort || Service.LOCKER_PORT;
```

Comment:
Shouldn't default ports for each service be extracted from `config.js` instead of hardcoded?